src/main.ts

```diff
@@ -22,19 +23,18 @@ async function run(): Promise<void> {
 
     const issueKey = githubConnector.getIssueKeyFromTitle();
 
-    if (!issueKey) {
-      console.log(`JIRA key was not found`);
-      process.exit(0);
-    }
+    if (!issueKey) throw core.error;
 
     console.log(`JIRA key -> ${issueKey}`);
 
-    const details = await jiraConnector.getTicketDetails(issueKey);
-    await githubConnector.updatePrDetails(details);
+    await jiraConnector.getTicketDetails(issueKey);
```

Comment:
please decouple these two methods back:
```
const details = await jiraConnector.getTicketDetails(issueKey);
await githubConnector.updatePrDetails(details);
```
1. It's important separation of concerns
2. in current approach you create GuthubConnector twice, which is useless

Reply:
These are now decoupled as per the original code.